use-cases/vehicle-routing/src/main/java/org/acme/vehiclerouting/domain/VehicleRoutePlan.java

```diff
@@ -0,0 +1,161 @@
+package org.acme.vehiclerouting.domain;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.stream.Stream;
+
+import ai.timefold.solver.core.api.domain.solution.PlanningEntityCollectionProperty;
+import ai.timefold.solver.core.api.domain.solution.PlanningScore;
+import ai.timefold.solver.core.api.domain.solution.PlanningSolution;
+import ai.timefold.solver.core.api.domain.solution.ProblemFactCollectionProperty;
+import ai.timefold.solver.core.api.domain.valuerange.ValueRangeProvider;
+import ai.timefold.solver.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import ai.timefold.solver.core.api.solver.SolverStatus;
+
+import org.acme.vehiclerouting.domain.geo.DrivingTimeCalculator;
+import org.acme.vehiclerouting.domain.geo.HaversineDrivingTimeCalculator;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+/**
+ * The plan for routing vehicles to customers, including:
+ * <ul>
+ *     <li>capacity - each vehicle has a capacity for customers demand,</li>
+ *     <li>time windows - each customer accepts the vehicle only in specified time window.</li>
+ * </ul>
+ *
+ * The planning solution is optimized according to the driving time (as opposed to the travel distance, for example)
+ * because it is easy to determine if the vehicle arrival time fits into the customer time window.
```

Comment:
I would even add one more sentence to explain that optimizing travel time optimizes the distance too, as a side effect, but in case there is a faster route, the travel time takes precedence (highway vs. some local road).

Reply:
Added.